overlay_new.py

```diff
@@ -32,10 +32,22 @@ def data_field_coord(x, y):
             """Coordinates of the data field in column x, row y."""
             return col_coords[x], row_coords[y]
 
+        # Dimensions of the left TransparentRectangle
+        left_rect = [
+            (0, row_coords[0] - DataField.height),
+            (col_coords[1] + DataField.width + spacing, self.height),
+        ]
+
+        # Dimensions of the right TransparentRectangle
+        right_rect = [
+            (col_coords[2], row_coords[0] - DataField.height),
+            (self.width, self.height),
+        ]
+
         # Create all overlay components
         self.components = [
-            TransparentRectangle((0, 560), (330, 740)),
-            TransparentRectangle((975, 560), (1280, 740)),
+            TransparentRectangle(left_rect[0], left_rect[1]),
+            TransparentRectangle(right_rect[0], right_rect[1]),
```

Comment:
Here's some syntax magic for you - basically it unpacks the array contents into function arguments.
```suggestion
            TransparentRectangle(*left_rect),
            TransparentRectangle(*right_rect),
```

Reply:
Woah thats actually really nifty